src/Strategy/Cargo.hs

```diff
@@ -156,6 +202,68 @@ instance ToJSON CargoProject
 instance AnalyzeProject CargoProject where
   analyzeProject _ = getDeps
 
+data CargoPackage = CargoPackage
+  { license :: Maybe Text.Text
+  , cargoLicenseFile :: Maybe (Path Rel File)
+  }
+  deriving (Eq, Show)
+
+cargoPackageCodec :: TomlCodec CargoPackage
+cargoPackageCodec =
+  CargoPackage
+    <$> dioptional (Toml.text "license") .= license
+    <*> pathCodec
+  where
+    filePathCodec = dioptional (Toml.string "license-file") .= (fmap toFilePath . cargoLicenseFile)
+    pathCodec = (>>= parseRelFile) <$> filePathCodec
+
+-- |Representation of a Cargo.toml file. See
+-- [here](https://doc.rust-lang.org/cargo/reference/manifest.html)
+-- for a description of this format.
+newtype CargoToml = CargoToml
+  {cargoPackage :: CargoPackage}
+  deriving (Eq, Show)
+
+cargoTomlCodec :: TomlCodec CargoToml
+cargoTomlCodec = diwrap (Toml.table cargoPackageCodec "package")
+-- ^^ The above is a bit obscure. It's generating a TomlCodec CargoPackage and
+-- then using 'diwrap'/Coercible to make a TomlCodec CargoToml.  I can't use
+-- 'CargoToml <$>' because TomlCodec aliases (Codec a a) and only (Codec a)
+-- has a Functor instance, so I'd end up with a (Codec CargoPackage CargoToml).
+
+instance LicenseAnalyzeProject CargoProject where
+  licenseAnalyzeProject = analyzeLicenses . cargoToml
+
+-- |Analyze a Cargo.toml for license information. The format is documented
+-- (here)[https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields]
+analyzeLicenses :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m [LicenseResult]
+analyzeLicenses tomlPath = do
+  pkg <- cargoPackage <$> readContentsToml cargoTomlCodec tomlPath
+  -- The license-file field in Cargo.toml is relative to the dir of the
+  -- Cargo.toml file. Generate an absolute path to license-file.
+  let maybeLicense = license pkg
+  let licensePathText = prependTomlDir <$> cargoLicenseFile pkg
+  let licenseCon = selectLicenseCon <$> maybeLicense
+  pure
+    [ LicenseResult
+        { licenseFile = toFilePath tomlPath
+        , licensesFound =
+            catMaybes
+              [ License <$> licenseCon <*> maybeLicense
+              , License LicenseFile <$> licensePathText
+              ]
+        }
+    ]
+  where
+    prependTomlDir = toText . toFilePath . (parent tomlPath </>)
+    textElem c = isJust . Text.findIndex (== c)
+    -- Old versions of Cargo allow '/' as a separator between SPDX values in
+    -- 'license'. In that case 'license' can't be treated as a LicenseSPDX.
+    selectLicenseCon licenseText =
+      if textElem '/' licenseText
```

Comment:
Use `Data.Text.elem` instead.
```suggestion
    -- Old versions of Cargo allow '/' as a separator between SPDX values in
    -- 'license'. In that case 'license' can't be treated as a LicenseSPDX.
    selectLicenseCon licenseText =
      if Text.elem '/' licenseText
```

Reply:
`Data.Text.elem` only exists starting in text version 1.2.5 as far as I can tell, and fossa-cli uses `^>= 1.2.3`, I did this to avoid upgrading. I can add that upgrade to this PR if you think it's reasonable.

Reply:
Damn.  We should probably upgrade some of our deps soon, this has hit us a few times recently.  Please leave it for now.